### ISSUE TITLE
Bitwarden Unified - Nginx - Permission denied

### ISSUE THREAD

**Issue by chrisonline** [bitwarden/server#2567]
### Steps To Reproduce

I used the below compose file on a Synology NAS with Portainer installed.
After deployment both containers are deployed and are running, but it is not working. The website it not shown.

Inside the log file nginx.log I see this error:
```
nginx: [emerg] mkdir() "/var/lib/nginx/tmp/client_body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/tmp/client_body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/tmp/client_body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/tmp/client_body" failed (13: Permission denied)
```
Log file of the container bitwarden:
```
Generating a RSA private key
............................++++
.................................................................................................................................................++++
writing new private key to '/etc/bitwarden/identity.key'
-----
2023-01-11 15:15:48,946 INFO Included extra file "/etc/supervisor.d/admin.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/api.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/events.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/icons.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/identity.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/nginx.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/notifications.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/scim.ini" during parsing
2023-01-11 15:15:48,947 INFO Included extra file "/etc/supervisor.d/sso.ini" during parsing
2023-01-11 15:15:48,962 INFO RPC interface 'supervisor' initialized
2023-01-11 15:15:48,962 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2023-01-11 15:15:48,963 INFO supervisord started with pid 52
2023-01-11 15:15:49,966 INFO spawned: 'identity' with pid 53
2023-01-11 15:15:49,969 INFO spawned: 'admin' with pid 54
2023-01-11 15:15:49,972 INFO spawned: 'api' with pid 55
2023-01-11 15:15:49,976 INFO spawned: 'icons' with pid 56
2023-01-11 15:15:49,979 INFO spawned: 'nginx' with pid 57
2023-01-11 15:15:49,985 INFO spawned: 'notifications' with pid 58
2023-01-11 15:15:50,151 INFO exited: nginx (exit status 1; not expected)
2023-01-11 15:15:51,155 INFO spawned: 'nginx' with pid 114
2023-01-11 15:15:52,188 INFO exited: nginx (exit status 1; not expected)
2023-01-11 15:15:54,819 INFO spawned: 'nginx' with pid 151
2023-01-11 15:15:55,599 INFO exited: nginx (exit status 1; not expected)
2023-01-11 15:15:58,607 INFO spawned: 'nginx' with pid 154
2023-01-11 15:15:58,625 INFO exited: nginx (exit status 1; not expected)
2023-01-11 15:15:59,627 INFO gave up: nginx entered FATAL state, too many start retries too quickly
2023-01-11 15:16:05,635 INFO success: identity entered RUNNING state, process has stayed up for > than 15 seconds (startsecs)
2023-01-11 15:16:05,635 INFO success: admin entered RUNNING state, process has stayed up for > than 15 seconds (startsecs)
2023-01-11 15:16:05,635 INFO success: api entered RUNNING state, process has stayed up for > than 15 seconds (startsecs)
2023-01-11 15:16:05,635 INFO success: icons entered RUNNING state, process has stayed up for > than 15 seconds (startsecs)
2023-01-11 15:16:05,636 INFO success: notifications entered RUNNING state, process has stayed up for > than 15 seconds (startsecs)
```

Used composed file with environment variables:
```
---
version: "3.8"

services:
  bitwarden:
    depends_on:
      - db
    image: bitwarden/self-host:beta
    restart: always
    ports:
      - "40009:8080"
      - "40010:8443"
    volumes:
      - /volume2/docker/bitwarden:/etc/bitwarden
      - /volume2/docker/bitwarden/logs:/var/log/bitwarden
    environment:
      PUID: 1026
      PGID: 100      
      BW_DOMAIN: "bitwarden.chrisonline.at"
      BW_DB_PROVIDER: mysql
      BW_DB_SERVER: db
      BW_DB_DATABASE: "bitwarden_vault"
      BW_DB_USERNAME: "bitwarden"
      BW_DB_PASSWORD: "xxx"
      BW_INSTALLATION_ID: "xxx"
      BW_INSTALLATION_KEY: "xxx"

  # MariaDB Example
  db:
    environment:
      MARIADB_USER: "bitwarden"
      MARIADB_PASSWORD: "xxx"
      MARIADB_DATABASE: "bitwarden_vault"
      MARIADB_RANDOM_ROOT_PASSWORD: "true"
    image: mariadb:10
    restart: always
    volumes:
      - /volume2/docker/bitwarden/data:/var/lib/mysql

volumes:
  bitwarden:
  data:
  logs:
```

### Expected Result

The website of bitwarden should be shown.

### Actual Result

Blank website with: ERR_EMPTY_RESPONSE

### Githash Version

Not possible

### Environment Details

- Operating system: Synology NAS (Linux)
- Environment: Docker (Portainer)

### Database Image

mariadb:10

### Issue-Link

https://github.com/bitwarden/server/issues/2480

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Skimige** [comment]
Use volumes (instead of bind mount) might be helping

**Comment by chrisonline** [comment]
> Use volumes (instead of bind mount) might be helping

What exactly do you mean?
That I should use a volume for "/var/lib/nginx"?

**Comment by Skimige** [comment]
> What exactly do you mean? That I should use a volume for "/var/lib/nginx"?

... i.e. to not modify "volumes" part, and keep the official example:

```diff
14,15c14,15
<       - /volume2/docker/bitwarden:/etc/bitwarden
<       - /volume2/docker/bitwarden/logs:/var/log/bitwarden
---
>       - bitwarden:/etc/bitwarden
>       - logs:/var/log/bitwarden
38c38
<       - /volume2/docker/bitwarden/data:/var/lib/mysql
---
>       - data:/var/lib/mysql
```

Like you, I replaced docker volumes with bind mounts when I try to start unified server and failed. I didn't do careful diagnose like yours but changing back to docker volumes fixed the issue.

**Comment by hoewer** [comment]
Creating the affected directories manually seems to solve the issue provisionally

`docker exec {Container name} mkdir /var/lib/nginx/tmp/client_body`
`docker exec {Container name} mkdir /var/lib/nginx/tmp/proxy`
`docker exec {Container name} mkdir /var/lib/nginx/tmp/fastcgi`
`docker exec {Container name} mkdir /var/lib/nginx/tmp/uwsgi`
`docker exec {Container name} mkdir /var/lib/nginx/tmp/scgi`

**Comment by kspearrin** [comment]
I think this is just an issue with the UID/GID you are setting in env vars. The `beta` tag that you are referencing is still using `GID` and `UID` so you will need to change those names. In the `dev` tag we have switched it to `PUID` and `PGID`.

**Comment by chrisonline** [comment]
Have now tried it with GID and UID instead PUIG and PGID and I have the same problem.
Currently, I fix this with manual creation of the directories, as @hoewer suggested.

**Comment by hoewer** [comment]
Without deeper examination I would venture the wild guess that it's just a timing issue. The creation of the directories by nginx is attempted before the assignment of the necessary permissions has been completed. A simple check on these permissions before starting nginx would help in this case...

**Comment by kspearrin** [comment]
Possible fix here: https://github.com/bitwarden/server/pull/2651

Can you try the latest `dev` tag and see if the issue is resolved? Be sure to set `PUID` and `PGID`.

**Comment by chrisonline** [comment]
> Possible fix here: #2651
> 
> Can you try the latest `dev` tag and see if the issue is resolved? Be sure to set `PUID` and `PGID`.

Seems it is fixed in the `dev` tag.
Thanks.
